engine/classes/ElggCrypto.php

```diff
@@ -25,15 +25,24 @@ class ElggCrypto {
 	/**
 	 * @var SiteSecret
 	 */
-	private $siteSecret;
+	private $site_secret;
 
 	/**
 	 * Constructor
 	 *
-	 * @param SiteSecret $siteSecret Secret service
+	 * @param SiteSecret $site_secret Secret service
+	 * @throws LoginException
 	 */
-	public function __construct(SiteSecret $siteSecret) {
-		$this->siteSecret = $siteSecret;
+	public function __construct(SiteSecret $site_secret = null) {
+		if (!$site_secret) {
```

Comment:
Eh, I'll just remove this change. This class has always been private API. It's tough for plugins who go ahead and use it.